npm/private/npm_translate_lock.bzl

```diff
@@ -447,10 +455,23 @@ def _impl(rctx):
         root_package = lock_attr.package
         link_workspace = lock_attr.workspace_name
 
-        lockfile_description = "@{} and @{}".format(
-            str(rctx.attr.package_json),
-            str(lock_attr),
+        lockfile_description = "{} and {}".format(
+            _label_str(rctx.attr.package_json),
+            _label_str(lock_attr),
+        )
+
+    # in Bazel 5.3.0, the lock file workspace_name will now be empty if it is the local workspace;
+    # we use rctx.attr.link_workspace instead to handle the explicit link workspace name case (e2e/rules_foo for example);
+    # check that there isn't conflicting link workspaces if both are set
+    if link_workspace and rctx.attr.link_workspace and link_workspace != rctx.attr.link_workspace:
```

Comment:
Without the new attribute there is no way to get the workspace name when vendoring 🙄

Even from the label `@rules_foo//:pnpm-lock.yaml`, if you're calling it in `rules_foo` for the purposes of vendoring then label.workspace_name is an empty string.

Reply:
Bazel strips the explicit workspace name from the label if it matches the user workspace name. This will make it non-breaking for users. It should only affect rule sets and libraries that want to vendor the resulting generated starlark.